src/libs/elektra/kdb.c

```diff
@@ -1006,6 +1007,14 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
 		goto error;
 	}
 
+	// TOOD (atmaxinger): improve
+	if (initHooks (handle, ksDup (elektraKs), handle->modules, errorKey) == -1)
```

Comment:
`elektraKs` is not the correct parameter for `KeySet * config`. Since `config` is passed right through to `elektraPluginOpen`, it is supposed to be the config for the plugin you're loading, i.e. needs to be different for every plugin.

However, AFAIK none of current global plugins require any config, so you can just always pass `ksNew(0, KS_END)` to `elektraPluginOpen`.